IPython/core/display.py

```diff
@@ -640,8 +640,11 @@ def __init__(self, data=None, url=None, filename=None, expanded=False, metadata=
         metadata: dict
             Specify extra metadata to attach to the json display object.
         """
-        self.expanded = expanded
-        self.metadata = metadata
+        self.metadata = {'expanded': expanded}
+        if metadata:
+            self.metadata.update(metadata)
+        if kwargs:
+            self.metadata.update(kwargs)
```

Comment:
I've moved the metadata merging logic from `_data_and_metadata` to init because I don't think`metadata` or `expanded` can change after init.